portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -0,0 +1,339 @@
+                                        if (!from.equals(to)) {
+                                            Session session = sessionManager.getCurrentSession();
+                                            String oldUserId = session.getUserId();
+                                            session.setUserId(to);
+                                            String url = assignmentService.getDeepLink(siteId, assignmentId);
+                                            sqlService.dbInsert(null
+                                                , ACADEMIC_INSERT_SQL
+                                                , new Object[] {from,to,event,ref,title,siteId,e.getEventTime(),url}
+                                                , "ID");
+                                            countCache.remove(to);
+                                        }
+                                    }
+                                }
+                            } catch (IdUnusedException idue) {
+                            }
+                        } else if (AssignmentConstants.EVENT_GRADE_ASSIGNMENT_SUBMISSION.equals(event)) {
+                            String siteId = pathParts[3];
+                            String submissionId = pathParts[pathParts.length - 1];
+                            try {
+                                SecurityAdvisor sa = unlock();
+                                AssignmentSubmission submission = assignmentService.getSubmission(submissionId);
+                                lock(sa);
+                                if (submission.getGradeReleased()) {
+                                    Site site = siteService.getSite(siteId);
+                                    Assignment assignment = submission.getAssignment();
+                                    String title = assignment.getTitle();
+                                    for (String to : ((List<String>) submission.getSubmitterIds())) {
+                                        Session session = sessionManager.getCurrentSession();
+                                        String oldUserId = session.getUserId();
+                                        session.setUserId(to);
+                                        String url = assignmentService.getDeepLink(siteId, assignment.getId());
+                                        sqlService.dbInsert(null
+                                            , ACADEMIC_INSERT_SQL
+                                            , new Object[] {from,to,event,ref,title,siteId,e.getEventTime(),url}
+                                            , "ID");
+                                        countCache.remove(to);
+                                    }
+                                }
+                            } catch (IdUnusedException idue) {
+                            }
+                        }
+                    }
+                } catch (Exception ex) {
+                    ex.printStackTrace();
```

Comment:
Not sure how this works with the `@Slf4j` annotation, do you need to use a logger here?